lib/PoolPlus.js

```diff
@@ -44,9 +44,11 @@ function validateMigrationStrategy(strategy) {
 class PoolPlus extends Pool {
   constructor(config) {
     const plusOptions = config.plusOptions || {};
+    delete config.plusOptions;
     validateMigrationStrategy(plusOptions.migrationStrategy);
 
-    super({config: new PoolConfig(config)});
+    const poolConfig = mysql.createPool(config).config;
+    super({config: poolConfig});
```

Comment:
It looks like you're trying to be less hacky, but this is going to create an extra, unused pool. This code doesn't need to change if you do the suggestion I'll post above.